### PR TITLE
Feature/cors allow credentials

### PR DESCRIPTION
**Pull request by cast**
Relates to issue https://github.com/awslabs/serverless-application-model/issues/463

Added support for CORS headers Access-Control-Expose-Headers and Access-Control-Allow-Credentials in the CORS Configuration block at the Serverless::Api resource.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/translator/output/api_with_cors_and_only_credentials.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                    "application/json": "{\n  \"statusCode\" : 200\n}\n"
+                  }, 
+                  "responses": {
+                    "default": {
+                      "statusCode": "200", 
+                      "responseTemplates": {
+                        "application/json": "{}\n"
+                      }, 
+                      "responseParameters": {
+                        "method.response.header.Access-Control-Allow-Origin": "'*'",
```
Comment (chrisoverzero) [inline tests/translator/output/api_with_cors_and_only_credentials.json]: This is invalid output. A-C-A-O isn't allowed to have the wildcard value when A-C-A-C is set.

Comment (brettstack) [reply tests/translator/output/api_with_cors_and_only_credentials.json]: Good catch! @cast could you add some logic to throw `InvalidDocument` when a user tries to specify `Allow-Credentials: true` and `Allow-Origin: *`?

**Comment by brettstack** [comment]
Thanks for raising this @chrisoverzero. After reviewing the spec and other documentation more thoroughly it does appear `Expose-Headers` is not intended for preflight responses.

**Comment by brettstack** [comment]
@cast if you can provide a repro/use-case for including this in preflight I can take another look. Otherwise the `Allow-Credentials` part of this PR is still very useful.

**Comment by cast** [comment]
Thanks @chrisoverzero for the spec review. Apart from a now fixed side effect leading me to think the allowed exposed headers was an issue in our internal development I added it mainly because other api management solutions often allow or force to take a liberal approach to preflight CORS (ApiMan, Axway,...). 

I'm not aware of specific browsers and browser versions that require the exposed header in the preflight response. I'm assuming if there is an issue it would be in old versions due to the current living standard.

I'll remove the exposed headers parts.

**Comment by JenningsWu** [comment]
Any updates? This is really helpful for our project : )

**Pull request merged**